### DISCUSSION TITLE
Custom picker mappings?

### DISCUSSION THREAD

**Discussion by jerrykan** [nvim-telekasten/telekasten.nvim#319] [Q&A] [answered]
Is there a way to set custom mappings for the telekasten pickers?

For the default telescope pickers I have mapped `<tab>` to `move_selection_next` and `<s-tab>` to `move_selection_previous`. However when pressing `<tab>` with telekasten `find_notes` picker it seems to insert a link to the first highlighted note in picker instead of using the default telescope behaviour.

I'm wonder if it is possible to have a custom mapping for the telekasten pickers that matches the custom mappings I have for the telescope pickers?

**Comment by lambtho12** [comment, marked as answer]
> I'm wonder if it is possible to have a custom mapping for the telekasten pickers that matches the custom mappings I have for the telescope pickers?

Not at the moment. It's been a while since I look at mappings, but my understanding is that we load the default telescope mappings and add ours on top. I suppose it could be possible to first load the current telescope mappings instead, but then you may have mapping clashes with the ones we provide for telekasten and that are currently fixed.

The ideal solution would be obviously to 
1. Always load telescope current mappings
2. Let the user change mappings for telekasten pickers
3. (Inform users when there is a mapping clash between telekasten and telescope)

I do not have the time to look into that now. This is something that should be part of a big configuration workflow refactor IMO. If you want to tackle this, you can open an issue and work on a PR, I will gladly find time to follow the development and review it.

**Reply by jerrykan** [reply]
@lambtho12 Thanks for the info.

It isn't a bit deal. I'm just starting to test out telekasten and was wondering if I was missing something obvious or not.
